no arguments to Optional type leads to index error

```py
# bar.py
from typing import Optional
def foo():
    # type: () -> Optional
    pass
```
```
$ mypy --fast-parser ./bar.py
...
    self.analyze_function(defn)
  File "/srv/zulip-venv-cache/cd17c4e82877061676869fc515f46b64bf073fa4/zulip-py3-venv/lib/python3.4/site-packages/mypy/semanal.py", line 468, in analyze_function
    defn.type = self.anal_type(defn.type)
  File "/srv/zulip-venv-cache/cd17c4e82877061676869fc515f46b64bf073fa4/zulip-py3-venv/lib/python3.4/site-packages/mypy/semanal.py", line 1038, in anal_type
    return t.accept(a)
  File "/srv/zulip-venv-cache/cd17c4e82877061676869fc515f46b64bf073fa4/zulip-py3-venv/lib/python3.4/site-packages/mypy/types.py", line 496, in accept
    return visitor.visit_callable_type(self)
  File "/srv/zulip-venv-cache/cd17c4e82877061676869fc515f46b64bf073fa4/zulip-py3-venv/lib/python3.4/site-packages/mypy/typeanal.py", line 182, in visit_callable_type
    ret_type=t.ret_type.accept(self),
  File "/srv/zulip-venv-cache/cd17c4e82877061676869fc515f46b64bf073fa4/zulip-py3-venv/lib/python3.4/site-packages/mypy/types.py", line 111, in accept
    return visitor.visit_unbound_type(self)
  File "/srv/zulip-venv-cache/cd17c4e82877061676869fc515f46b64bf073fa4/zulip-py3-venv/lib/python3.4/site-packages/mypy/typeanal.py", line 114, in visit_unbound_type
    return items[0]
IndexError: list index out of range

*** INTERNAL ERROR ***

bar.py:2: error: Internal error -- please report a bug at https://github.com/python/mypy/issues

NOTE: you can use "mypy --pdb ..." to drop into the debugger when this happens.
```

--- a/mypy/test/data/semanal-types.test
+++ b/mypy/test/data/semanal-types.test
@@ -1434,7 +1434,7 @@ MypyFile:1(
 [case testInvalidOptionalType]
 from typing import Optional
 x = 1  # type: Optional[int, str]  # E: Optional[...] must have exactly one type argument
-y = 1  # type: Optional[x]  # E: Invalid type "__main__.x"
+y = 1  # type: Optional  # E: Optional[...] must have exactly one type argument
 [out]
 
 [case testCoAndContravariantTypeVar]

--- a/mypy/typeanal.py
+++ b/mypy/typeanal.py
@@ -108,6 +108,7 @@ class TypeAnalyser(TypeVisitor[Type]):
             elif fullname == 'typing.Optional':
                 if len(t.args) != 1:
                     self.fail('Optional[...] must have exactly one type argument', t)
+                    return AnyType()
                 items = self.anal_array(t.args)
                 # Currently Optional[t] is just an alias for t.
                 return items[0]

